### ISSUE TITLE
rake-10.5.0/bin/rake does not exist error message when upgrading to 2.7.2

### ISSUE THREAD

**Issue by wlipa** [rubygems/rubygems#2071]
When updating rubygems from 2.6.14 to 2.7.2, I see the following error message about an old version of rake:

```
$ sudo gem update --system
[sudo] password for ops: 
Updating rubygems-update
Fetching: rubygems-update-2.7.2.gem (100%)
Successfully installed rubygems-update-2.7.2
Installing RubyGems 2.7.2
Bundler 1.16.0 installed
RubyGems 2.7.2 installed
Regenerating binstubs
`/usr/share/rubygems-integration/all/gems/rake-10.5.0/bin/rake` does not exist, maybe `gem pristine rake` will fix it?
RubyGems system software updated
```

The install seems to work though.

Here are my current environment details:

```
$ gem env version
RubyGems Environment:
  - RUBYGEMS VERSION: 2.7.2
  - RUBY VERSION: 2.3.1 (2016-04-26 patchlevel 112) [x86_64-linux-gnu]
  - INSTALLATION DIRECTORY: /var/lib/gems/2.3.0
  - USER INSTALLATION DIRECTORY: /home/ops/.gem/ruby/2.3.0
  - RUBY EXECUTABLE: /usr/bin/ruby2.3
  - EXECUTABLE DIRECTORY: /usr/local/bin
  - SPEC CACHE DIRECTORY: /home/ops/.gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: /etc
  - RUBYGEMS PLATFORMS:
    - ruby
    - x86_64-linux
  - GEM PATHS:
     - /var/lib/gems/2.3.0
     - /home/ops/.gem/ruby/2.3.0
     - /usr/lib/x86_64-linux-gnu/rubygems-integration/2.3.0
     - /usr/share/rubygems-integration/2.3.0
     - /usr/share/rubygems-integration/all
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => false
     - :bulk_threshold => 1000
     - "gem" => "--no-rdoc --no-ri"
     - "install" => "--user-install -n /home/ops/bin"
     - "update" => "--user-install -n /home/ops/bin"
  - REMOTE SOURCES:
     - https://rubygems.org/
  - SHELL PATH:
     - /home/ops/bin
     - /usr/local/sbin
     - /usr/local/bin
     - /usr/sbin
     - /usr/bin
     - /sbin
     - /bin
     - /usr/games
     - /usr/local/games
     - /snap/bin
     - ./bin
```

There are a few rake versions installed on this machine.  I don't know why the rubygems update process is generating a message about this old version of rake.
rake (12.2.1, 12.1.0, 12.0.0, 10.5.0)

The referenced directory indeed does not exist, because /usr/share/rubygems-integration/all/ only contains bin and specifications subdirs.

**Comment by hsbt** [comment]
Can you give a reproducible instruction?

**Comment by wlipa** [comment]
Install a plain vanilla Ubuntu 17.10 on, for example, Google Cloud Platform.  Then the following steps will show the rake error message:

    sudo apt install ruby2.3 ruby2.3-dev
    sudo gem update --system

In this particular case, the error looks like:

    `/usr/share/rubygems-integration/all/gems/rake-12.0.0/exe/rake` does not exist, maybe `gem pristine rake`

**Comment by hsbt** [comment]
Thanks.

Debian/Ubuntu platform modified the location of executable files. So, It's not defect for rubygems 2.7.x. We need to support or change the specification of `pristine` commands for debian/ubuntu platform.

**Comment by tjouan** [comment]
Didn't Debian use to disable this command in their packaged version?

IMO it would be more safe and manageable to disable it and just
rebuild your ruby and/or a rubygems package with updated version (I
used to proceed this way when using Debian some time ago).

Or maybe not use the package at all and make a custom user install
from source as last resort, might be less maintainable so I wouldn't
recommend, but maybe for specify use cases it would work.

**Comment by hsbt** [comment]
This may be fixed with https://github.com/rubygems/rubygems/pull/2208 and other fixes with 2.7.3+.
